apps/files/src/views/TemplatePicker.vue

```diff
@@ -213,19 +213,27 @@ export default {
 				)
 				this.logger.debug('Created new file', fileInfo)
 
-				await fileList?.addAndFetchFileInfo(this.name)
-
-				// Run default action
 				const fileAction = OCA.Files.fileActions.getDefaultFileAction(fileInfo.mime, 'file', OC.PERMISSION_ALL)
-				fileAction.action(fileInfo.basename, {
-					$file: fileList?.findFileEl(this.name),
-					dir: currentDirectory,
-					fileList,
-					fileActions: fileList?.fileActions,
-					fileInfoModel: fileList?.getModelForFile(this.name),
-				})
 
-				this.close()
+				fileList?.addAndFetchFileInfo(this.name).then((status, data) => {
```

Comment:
Not pretty, but might simply the code:

```suggestion
                // Use an intermediary `then` to bundle 'status' and 'data' in one destructurable object.
				const { status, data } = await fileList?.addAndFetchFileInfo(this.name).then((status, data) => ({status, data}))
```

Reply:
@artonge That's exactly what I was looking for. Thank you dear JS ninja :grin:.
Rebased on master. Now uses this shorter and (almost) elegant solution.